### DISCUSSION TITLE
OTP login (passwordless login)

### DISCUSSION THREAD

**Discussion by aasutossh** [pocketbase/pocketbase#4828] [Q&A]
Is it possible to achieve this?
Kind of custom authentication.

Flow is like =>
1. User enters their phone
2. We send the OTP to their phone, and keep track of the OTP in the user table
3. If user enter the correct OTP, we authenticate the user as authenticated user.

**Comment by ganigeorgiev** [comment]
2FA and OTP (_via email code; SMS code can be send by intercepting the OTP email hook_) are already implemented as part of the [ongoing refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) (_there are no ETAs yet_).

If you want to implement it on your own, you can have a look at https://github.com/pocketbase/pocketbase/issues/119#issuecomment-1881906233.
